File: src/main/java/com/bvanseg/gigeresque/client/entity/model/FacehuggerEntityModel.java
package com.bvanseg.gigeresque.client.entity.model;

import com.bvanseg.gigeresque.client.entity.animation.EntityAnimations;
import com.bvanseg.gigeresque.client.entity.texture.EntityTextures;
import com.bvanseg.gigeresque.common.entity.impl.FacehuggerEntity;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.util.Identifier;
import software.bernie.geckolib3.model.AnimatedGeoModel;

@Environment(EnvType.CLIENT)
public class FacehuggerEntityModel extends AnimatedGeoModel<FacehuggerEntity> {
    @Override
    public Identifier getModelLocation(FacehuggerEntity object) {
        return EntityModels.FACEHUGGER;
    }

    @Override
    public Identifier getTextureLocation(FacehuggerEntity object) {
        return EntityTextures.FACEHUGGER;
    }

    @Override
    public Identifier getAnimationFileLocation(FacehuggerEntity animatable) {
        return EntityAnimations.FACEHUGGER;
    }
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/model/FacehuggerEntityModel.kt
package com.bvanseg.gigeresque.client.entity.model

import com.bvanseg.gigeresque.client.entity.animation.EntityAnimations
import com.bvanseg.gigeresque.client.entity.texture.EntityTextures
import com.bvanseg.gigeresque.common.entity.impl.FacehuggerEntity
import net.minecraft.util.Identifier
import software.bernie.geckolib3.model.AnimatedGeoModel

/**
 * @author Boston Vanseghi
 */
class FacehuggerEntityModel : AnimatedGeoModel<FacehuggerEntity>() {
    override fun getModelLocation(entity: FacehuggerEntity): Identifier = EntityModels.FACEHUGGER
    override fun getTextureLocation(entity: FacehuggerEntity): Identifier = EntityTextures.FACEHUGGER
    override fun getAnimationFileLocation(animatable: FacehuggerEntity): Identifier = EntityAnimations.FACEHUGGER
}